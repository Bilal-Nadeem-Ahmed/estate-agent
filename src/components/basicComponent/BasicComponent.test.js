import React from "react";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";

import BasicComponent from "./index";

let container = null;

describe("<BasicComponent />", () => {
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  it("renders component without any props", () => {
    act(() => {
      render(<BasicComponent />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
  });

  it("renders component with name = Bilal", () => {
    act(() => {
      render(<BasicComponent name="Bilal" />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
  });
});
