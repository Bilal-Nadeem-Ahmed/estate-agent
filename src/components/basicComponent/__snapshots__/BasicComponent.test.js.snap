// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicComponent /> renders component with name = Bilal 1`] = `"<div>Hello, Bilal</div>"`;

exports[`<BasicComponent /> renders component without any props 1`] = `""`;
